oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/MoveVersionableNodeRepositoryTest.java

```diff
@@ -0,0 +1,229 @@
+        if (newNode.canAddMixin(JcrConstants.MIX_VERSIONABLE)) {
+            newNode.addMixin(JcrConstants.MIX_VERSIONABLE);
+        } else {
+            throw new NotExecutableException();
+        }
+        newNode.getSession().save();
+        return newNode;
+    }
+
+    /**
+     * Checks out the node, sets the property then saves the session and checks the node back in.
+     * To be used in tests where version history needs to be populated.
+     */
+    private void setNodePropertyAndCheckIn(Node node, String propertyName, String propertyValue) throws Exception {
+        node.checkout();
+        node.setProperty(propertyName, propertyValue);
+        node.getSession().save();
+        node.checkin();
+    }
+
+    /*
+     * 1. Create a versionable unstructured node at nodeName1/nodeName2/sourceNode
+     * 2. Create a versionable unstructured node at nodeName1/nodeName3/sourceNode
+     * 3. create version histories for both nodes
+     * 4. remove nodeName1/nodeName3/nodeName1 (that's because move(src,dest) throws an exception if dest already exists)
+     * 5. move nodeName1/nodeName2/sourceNode to nodeName1/nodeName3/sourceNode and call session.save()
+     * 6. should work according to JCR specification - reproduces bug for OAK-8848: will throw ConstraintViolationException
+     *  - "Property is protected: jcr:versionHistory
+     */
+    @Test
+    public void testMoveNodeWithVersionHistoryOverDeletedNodeWithVersionHistory() throws Exception {
+
+        Session session = getAdminSession();
+        Node testRootNode = session.getRootNode().addNode("node1");
+
+        String newNodeName = "sourceNode";
+        Node sourceParent = testRootNode.addNode("node2"); // nodeName1/nodeName2
+        Node sourceNode = createVersionableNode(sourceParent, newNodeName); // nodeName1/nodeName2/sourceNode
+        Node destParent = testRootNode.addNode("node3"); // nodeName1/nodeName3
+        Node destNode = createVersionableNode(destParent, "destNode"); // nodeName1/nodeName3/sourceNode
+
+        String destPath = destNode.getPath();
+
+        // add version histories for sourceNode and destNode
+        setNodePropertyAndCheckIn(sourceNode, "sourceNode_testProp", "sourceNode_testValue_1");
+
+        session.save();
+
+        setNodePropertyAndCheckIn(sourceNode, "sourceNode_testProp", "sourceNode_testValue_2");
+        setNodePropertyAndCheckIn(destNode, "destNode_testProp", "destNode_testValue_2");
+
+        session.save();
+
+        VersionHistory delNodeVersionHistory = destNode.getVersionHistory();
+        String delNodeVHPath = delNodeVersionHistory.getPath();
+
+        session.removeItem(destNode.getPath());
+
+        session.move(sourceNode.getPath(), destPath);
+        session.save();
+
+        // check setting property - via NodeImpl - on moved node
+        setNodePropertyAndCheckIn(sourceNode, "testProp", "testValue");
+
+        // check version history - TODO HORIA: should be in separate method - EXTRACT
```

Comment:
I would avoid personal TODOs and just have it without username..

Reply:
Fixed it, thank you for the catch.